### DISCUSSION TITLE
How to apply gradient on the opposite side of the AreaChart line

### DISCUSSION THREAD

**Discussion by nikmace** [recharts/recharts#4559] [Q&A] [answered]
## Do you want to request a feature or report a bug?
**Feature**

Sorry for creating an issue, was not sure where to ask otherwise. Thank you in advance!

## What is the current behavior?
I can only apply the gradient below the line.

Here is my AreaChart:

```html
       <ResponsiveContainer>
          <AreaChart
            syncId="depthChartSync"
            data={processedAsks}
            margin={{ top: 10, right: 20, left: 15, bottom: 0 }}
            onMouseMove={(e) => {
              setActiveIndex(e.activeTooltipIndex!);
            }}
            onMouseLeave={() => {
              setActiveIndex(-1);
            }}
          >
            <defs>
              <linearGradient id="colorAsks" x1="0" y1="0" x2="0" y2="1">
                <stop offset="0%" stopColor="#eb5757" stopOpacity={0.01} />
                <stop offset="100%" stopColor="#eb5757" stopOpacity={0.3} />
              </linearGradient>
            </defs>

            <XAxis
              dataKey="volume"
              tickCount={5}
              axisLine={false}
              tickLine={false}
              tick={false}
              domain={asksVolumeDomain}
              type="number"
            />
            <YAxis
              dataKey="price"
              orientation="right"
              domain={asksDomain}
              axisLine={{ stroke: '#eb575789', strokeWidth: 1 }}
              tickLine={false}
              tick={tickStyles}
              type="number"
              tickFormatter={(value) =>
                `$${value.toLocaleString('en-US', { minimumFractionDigits: 0, maximumFractionDigits: 2 })}`
              }
              tickCount={5}
            />
            <Tooltip cursor={false} content={<></>} />
            <Area
              type="monotone"
              dataKey="price"
              stroke="#eb5757"
              fillOpacity={1}
              fill="url(#colorAsks)"
              dot={false}
              activeDot={
                // @ts-ignore
                <CustomScatterDot
                  chartWidth={chartWidth}
                  accentColor="#F23645"
                  secondaryColor="#90252d"
                  activeIndex={activeIndex}
                  bidsData={processedBids}
                  asksData={processedAsks}
                  type="asks"
                />
              }
            />
          </AreaChart>
        </ResponsiveContainer>
```
Currently it looks like this:

<img width="363" alt="Screenshot 2024-05-18 at 17 12 23" src="https://github.com/recharts/recharts/assets/39880364/65fcd667-2665-4d61-96f0-6a642938466b">

## What is the expected behavior?
The gradient appears on the opposite side of the line, meaning from the line to the top of the chart.

And I want to make the gradient like this:
(so it appears where I have drawn white lines, so it is inverted in relation to the line)
<img width="363" alt="Screenshot 2024-05-18 at 17 12 23" src="https://github.com/recharts/recharts/assets/39880364/520e02ea-2e47-4a52-b83d-9c364ece6c37">

## Which versions of Recharts, and which browser / OS are affected by this issue? Did this work in previous versions of Recharts?
recharts: 2.12.1
Google Chrome
Not sure if this ever worked

**Comment by ckifer** [comment]
Hi!

There's nothing rendered on the other side of the area, so there's nothing to apply a gradient to. We'd have to support inverting the whole chart I think which isn't currently possible. 

There might be some way to make it work without that kind of change but off the top of my head I can't think of anything.

**Comment by ckifer** [comment]
Going to move this to a discussion - if you'd like to request a feature, feel free to create another issue. I'm not sure this is possible in any charting library, feel free to share examples if it is

**Comment by spennyp** [comment]
See that sometimes its above, sometimes its below (depending on values):
<img width="569" alt="Screenshot 2024-05-24 at 23 58 10" src="https://github.com/recharts/recharts/assets/31705429/12b11af5-34ed-4817-85c6-53405dbda4ae">

Seems there is currently no to control where it appears?

**Reply by ckifer** [reply, marked as answer]
Sure there is control, there is the `baseValue` prop

It defaults to 0, so normally anything under 0 will be filled above the line. Anything above 0 will be filled below the line.

If I change the `baseValue` then that becomes the new point it changes
![image](https://github.com/recharts/recharts/assets/25180830/ee7e70e5-9fa0-47d3-bfb9-9439304dd9bd)

**Comment by ckifer** [comment]
@nikmace Try changing `baseValue` - it would have to be beyond your maximum datapoint, but it should then fill from that line down instead of 0 up. I didn't understand until @spennyp's post. Let me know if that resolves it

**Comment by nikmace** [comment]
Thank you, guys, for your responses, much appreciated! :)

Here is the end result, exactly what I wanted to achieve:

<img width="373" alt="Screenshot 2024-05-27 at 9 24 42" src="https://github.com/recharts/recharts/assets/39880364/69e1ddab-4d86-4858-aa8a-c1f4634b9ca5">

For those struggling to find a solution with dynamic data, when you calculate the **domain** for the **YAxis** (here YAxis is the price) (the `min` and the `max` value) you can set the `baseValue` to the `max value` from the YAxis domain.

**Reply by ckifer** [reply]
Nice!
